label.py

```diff
@@ -0,0 +1,93 @@
+import math
+from typing import TYPE_CHECKING, Callable, Optional, Union
+
+from SubpacketIDs import SubpacketEnum
+
+if TYPE_CHECKING:
+    from main import MainApp
+
+
+class Label:
+    def __init__(
+        self,
+        name: str,
+        update_func: Callable[["MainApp"], str],
+        display_name: Optional[str] = None,
+    ):
+        self.name = name
+        self.display_name = display_name if display_name is not None else name
+        self.update = update_func
+
+
+def update_altitude(main_window: "MainApp") -> str:
+    return str(main_window.data.lastvalue(SubpacketEnum.CALCULATED_ALTITUDE.value))
+
+
+def update_max_altitude(main_window: "MainApp") -> str:
+    return str(main_window.data.highest_altitude)
+
+
+def update_gps(main_window: "MainApp") -> str:
+    latitude = main_window.data.lastvalue(SubpacketEnum.LATITUDE.value)
+    longitude = main_window.data.lastvalue(SubpacketEnum.LONGITUDE.value)
+    return str(latitude) + ", " + str(longitude)
+
+
+def update_state(main_window: "MainApp") -> str:
+    return str(main_window.data.lastvalue(SubpacketEnum.STATE.value))
+
+
+def update_pressure(main_window: "MainApp") -> str:
+    return str(main_window.data.lastvalue(SubpacketEnum.PRESSURE.value))
+
+
+def update_acceleration(main_window: "MainApp") -> str:
+    def nonezero(x: Union[float, None]) -> float:
+        return 0 if x is None else x
+
+    accel = math.sqrt(
+        nonezero(main_window.data.lastvalue(SubpacketEnum.ACCELERATION_X.value)) ** 2
+        + nonezero(main_window.data.lastvalue(SubpacketEnum.ACCELERATION_Y.value)) ** 2
+        + nonezero(main_window.data.lastvalue(SubpacketEnum.ACCELERATION_Z.value)) ** 2
+    )
+    return str(accel)
+
+
+# TODO: Implement Tantalus test separation label update.
+def update_test_separation(main_window: "MainApp") -> str:
+    return "Separated"
+
+
+# TODO: Implement Co-Pilot tank pressure label update.
+def update_tank_pressure(main_window: "MainApp") -> str:
+    return "10 Pa"
+
+
+# TODO: Implement Co-Pilot chamber pressure label update.
+def update_chamber_pressure(main_window: "MainApp") -> str:
+    return "40 Pa"
+
+
+# TODO: Implement Co-Pilot chamber temperature label update.
+def update_chamber_temp(main_window: "MainApp") -> str:
+    return "283 K"
+
+
+default_labels = [
+    Label("Altitude", update_altitude),
+    Label("MaxAltitude", update_max_altitude, "Max Altitude"),
+    Label("GPS", update_gps),
+    Label("State", update_state),
+    Label("Pressure", update_pressure),
+    Label("Acceleration", update_acceleration),
+]
+
+tantalus_labels = default_labels + [
+    Label("TestSeparation", update_test_separation, "Test Separation")
+]
+
+cp_labels = default_labels + [
+    Label("TankPressure", update_tank_pressure, "Tank Pressure"),
+    Label("ChamberPressure", update_chamber_pressure, "Chamber Pressure"),
+    Label("ChamberTemp", update_chamber_temp, "Chamber Temperature"),
+]
```

Comment:
All the functions are nice to have here, but can we please move the lists into the rocket-specific files mentioned in the other comment? Having one file that describes everything for each rocket is nice because when someone eventually needs to change something they don't have to go digging through multiple files to find where it came from. 

Also, we can get rid of the `default_labels` list as it might not be very default for long and it will help with keeping everything for each rocket together.

Reply:
Done. The root directory of the repository is getting a bit hectic though. I'll likely submit another pull request that puts all of my new files into a "profiles" folder or something and do the same for some other files once this branch (and possibly Sepand's) gets pulled.

Reply:
That would be awesome if you do that. I see you already added yourself to https://trello.com/c/VY5osi76/230-reorganize-ground-station-directories , so I guess this fits within its scope nicely.